k8s/controllers/actions/catalogeval_controller.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	federationv1 "gopls-workspace/apis/federation/v1"
+
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// CatalogReconciler reconciles a Site object
+type CatalogEvalReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	// ApiClient is the client for Symphony API
+	ApiClient utils.ApiClient
+}
+
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/finalizers,verbs=update
+
+func (r *CatalogEvalReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ret ctrl.Result, reterr error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entering the CatalogEvalExpression reconciler")
```

Comment:
Please use new logger to log with context - https://github.com/eclipse-symphony/symphony/commit/578a5f9579280a355f3d4dd1b781d0116fe981a5#diff-be51259dbae1b4fa926a318e2ff8da00395f1028d32f1bd077acc40705a5a84f.